src/CookieCollection.php

```diff
@@ -0,0 +1,298 @@
+     * @return bool whether the named cookie exists
+     */
+    public function offsetExists($name): bool
+    {
+        return $this->has($name);
+    }
+
+    /**
+     * Returns the cookie with the specified name.
+     * This method is required by the SPL interface [[\ArrayAccess]].
+     * It is implicitly called when you use something like `$cookie = $collection[$name];`.
+     * This is equivalent to [[get()]].
+     * @param string $name the cookie name
+     * @return Cookie the cookie with the specified name, null if the named cookie does not exist.
+     */
+    public function offsetGet($name): Cookie
+    {
+        return $this->get($name);
+    }
+
+    /**
+     * Adds the cookie to the collection.
+     * This method is required by the SPL interface [[\ArrayAccess]].
+     * It is implicitly called when you use something like `$collection[$name] = $cookie;`.
+     * This is equivalent to [[add()]].
+     * @param string $name the cookie name
+     * @param Cookie $cookie the cookie to be added
+     */
+    public function offsetSet($name, $cookie): void
+    {
+        $this->add($cookie);
+    }
+
+    /**
+     * Removes the named cookie.
+     * This method is required by the SPL interface [[\ArrayAccess]].
+     * It is implicitly called when you use something like `unset($collection[$name])`.
+     * This is equivalent to [[remove()]].
+     * @param string $name the cookie name
+     */
+    public function offsetUnset($name): void
+    {
+        $this->remove($name);
+    }
+
+    /**
+     * Returns the number of cookies in the collection.
+     * This method is required by the SPL `Countable` interface.
+     * It will be implicitly called when you use `count($collection)`.
+     * @return int the number of cookies in the collection.
+     */
+    public function count(): int
+    {
+        return count($this->cookies);
+    }
+
+    /**
+     * Returns the cookie with the specified name.
+     * @param string $name the cookie name
+     * @return Cookie the cookie with the specified name. Null if the named cookie does not exist.
+     * @see getValue()
+     */
+    public function get(string $name): Cookie
+    {
+        return $this->cookies[$name];
+    }
+
+    /**
+     * Returns the value of the named cookie.
+     * @param string $name the cookie name
+     * @param mixed $defaultValue the value that should be returned when the named cookie does not exist.
+     * @return mixed the value of the named cookie.
+     * @see get()
+     */
+    public function getValue(string $name, $defaultValue = null): ?string
+    {
+        return isset($this->cookies[$name]) ? $this->cookies[$name]->getValue() : $defaultValue;
+    }
+
+    /**
+     * Adds a cookie to the collection.
+     * If there is already a cookie with the same name in the collection, it will be removed first.
+     * @param Cookie $cookie the cookie to be added
+     */
+    public function add(Cookie $cookie): void
```

Comment:
What's the goal of adding a cookie to it? Is `addToResponse(ResponseInterface $response): ResponseInterface` planned?

Reply:
Yep, will push the pending changes soon, they are almost ready. The class will have `addToResponse` and `setToResponse`, and it also already have `fromResponse`. So this is just a collection of cookies and easy way to handle (add, modify,...) all the cookies together when needed.